include/caf/click_detector.rst

```diff
@@ -0,0 +1,79 @@
+.. _caf_click_detector:
+
+CAF: Click detector module
+##########################
+
+.. contents::
+   :local:
+   :depth: 2
+
+The |click_detector| of the :ref:`lib_caf` (CAF) is responsible for generating a ``click_event`` when a known type of click is recorded for the button defined in the module configuration.
+
+Click type
+**********
+
+The module records the following click types:
+
+* :c:enumerator:`CLICK_SHORT` - Button pressed and released after short time.
+* :c:enumerator:`CLICK_NONE` - Button pressed and held for a period of time that is too long for :c:enumerator:`CLICK_SHORT`, but too short for :c:enumerator:`CLICK_LONG`.
+* :c:enumerator:`CLICK_LONG` - Button pressed and held for a long period of time.
+* :c:enumerator:`CLICK_DOUBLE` - Two sequences of the button press and release in a short time interval.
+
+The exact values of time intervals for click types are defined in the :file:`subsys/caf/modules/click_detector.c` file.
+
+Configuration
+*************
+
+The |click_detector| detects click types based on ``button_event``.
+For this reason, you must enable a module that generates ``button_event``.
+This event can be generated for example by the :ref:`caf_buttons` module.
+
+You must also add a configuration file for the module.
+The configuration file contains an array of :c:struct:`click_detector_config`.
+The array defines the following information for every click detector:
+
+* :c:member:`click_detector_config.key_id` - ID of the selected key.
+* :c:member:`click_detector_config.consume_button_event` - Whether the ``button_event`` with the given :c:member:`click_detector_config.key_id` should be consumed by the module.
+
+For example, the file content could look like follows:
+
+.. code-block:: c
+
+	static const struct click_detector_config click_detector_config[] = {
+		{
+			.key_id = KEY_ID(0x00, 0x01),
+			.consume_button_event = false,
+		},
+	};
+
+.. note::
+   The :c:macro:`KEY_ID` macro is defined in :file:`include/caf/key_ids.h`.
```

Comment:
Why don't you add `#include <caf/key_id.h>` in this code snippet? Btw: `s/key_ids/key_id`

Reply:
Added. Fixed.